cns/client/client.go

```diff
@@ -93,8 +94,60 @@ func buildRoutes(baseURL string, paths []string) (map[string]url.URL, error) {
 	return routes, nil
 }
 
-// GetNetworkConfiguration Request to get network config.
-func (c *Client) GetNetworkConfiguration(ctx context.Context, orchestratorContext []byte) (*cns.GetNetworkContainerResponse, error) {
+// GetAllNetworkContainers Request to get network container configs.
+func (c *Client) GetAllNetworkContainers(ctx context.Context, orchestratorContext []byte) ([]cns.GetNetworkContainerResponse, error) {
+	payload := cns.GetNetworkContainerRequest{
+		OrchestratorContext: orchestratorContext,
+	}
+
+	var body bytes.Buffer
+	if err := json.NewEncoder(&body).Encode(payload); err != nil {
+		return nil, &CNSClientError{
+			Code: types.UnexpectedError,
+			Err:  err,
+		}
+	}
+
+	u := c.routes[cns.GetAllNetworkContainers]
+	req, err := http.NewRequestWithContext(ctx, http.MethodPost, u.String(), &body)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to build request")
+	}
+	req.Header.Set(headerContentType, contentTypeJSON)
+	res, err := c.client.Do(req)
+	if err != nil {
+		return nil, errors.Wrap(err, "http request failed")
+	}
+	defer res.Body.Close()
+
+	if res.StatusCode != http.StatusOK {
```

Comment:
we should handle the case where the res.StatusCode is 404 indicating that the API is not supported.
Let's add a todo for CNS to respond with 405 if the API is not supported.
Can we add UnsupportedAPI error code in the types and return it similar to line 125?